### ISSUE TITLE
there so special char in the string 

### ISSUE THREAD

**Issue by linkt** [cpp-netlib/cpp-netlib#524]
hello, 
when I get a jsonstring from the Apache server I find that there are some other char at the start and end.

example:
{"responseHeader":{"status":0,"QTime":0,"params":{"requestType":5,"moduleType":2,"wt":"json","inputPath":"","outputPath":"","jobName":""}},"Response":{"numFound":0,"start":0,"docs":[{"return":"result:100\n"}]}}
it is the string that I get from the Wirshark.

d2

{"responseHeader":{"status":0,"QTime":0,"params":{"requestType":5,"moduleType":2,"wt":"json","inputPath":"","outputPath":"","jobName":""}},"Response":{"numFound":0,"start":0,"docs":[{"return":"result:100\n"}]}}

0
it get from the body 

ps: part of my codes:

```
std::string json_type = "application/x-www-form-urlencoded";
			http::client client;
			http::client::request req(url);
			req << boost::network::header("Content-Type", json_type);
			http::client::response response;
			std::string bodystr;
			bodystr.clear();
			try{
				 response = client.post(req, json_body, json_type);
				 if (body(response).size() == 0)
					 throw;
				 bodystr = body(response);
				 ofstream inf("test.txt");
				inf << bodystr;
			}
			catch (std::exception &e){
				shared_ptr<UnsupportOperationException> ptr = shared_ptr<UnsupportOperationException>(new UnsupportOperationException());
				bs->setException(ptr);
				std::cerr << e.what() << std::endl;
			}



```

**Comment by mikhailberis** [comment]
What version of the library are you using?

**Comment by linkt** [comment]
cpp-netlib-0.11.0

**Comment by linkt** [comment]
I try to use the cpp-netlib-11.1 ，and the json string is  wrong ,too.

**Comment by linkt** [comment]
cpp-netlib-0.11.0

**Comment by deanberris** [comment]
Please try 0.11.1 if you can, or the latest in 0.11-devel to see if it's still a problem?

**Comment by linkt** [comment]
I try to use the cpp-netlib-11.1 ，and the json string is still wrong

**Comment by deanberris** [comment]
This is probably caused by chunked encoding transfer. If you set the 'Connection' header to 'close', it should stop the server from sending you chunked transfer encoded responses. That said I thought the client already handles chunked transfer encoding properly.

I would still encourage you to try the latest in the 0.11-devel branch to see whether this hasn't already been fixed in the meantime.

**Comment by linkt** [comment]
yes ,I think you are right, this is the header of request which is coded using java,I find there are some 
diference between c++ and java.
I try to make the header of c++ being same to java : 
![image](https://cloud.githubusercontent.com/assets/7455401/7557272/dc5b3a3a-f7c0-11e4-9bd1-aed83c06ebf4.png)


 
I use the latest in the 0.11-devel branch. unfortunately,,The problem can't be fixed

**Comment by linkt** [comment]
I find the same issure
https://github.com/cpp-netlib/cpp-netlib/issues/110

**Comment by deanberris** [comment]
This is already fixed in 0.11.1 and 0.11-devel -- what you've shown here is that the Java implementation uses `Connection: Keep-Alive`. If you want the server to not use chunked transfer encoding, set `Connection: close` instead.

If your server doesn't respond with the `Transfer-Encoding: chunked` header, then the client will not treat it as something that has chunked transfer encoding.

Do you have a way to reproduce this with code, preferably making requests to public websites so I can try and debug this myself?

**Comment by deanberris** [comment]
This should now be fixed.
